org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/DefaultGradleBuild.java

```diff
@@ -22,28 +21,37 @@
 
 import org.eclipse.buildship.core.GradleBuild;
 import org.eclipse.buildship.core.SynchronizationResult;
+import org.eclipse.buildship.core.configuration.BuildConfiguration;
 import org.eclipse.buildship.core.internal.workspace.NewProjectHandler;
 
 /**
  * Default implementation for {@link GradleBuild}.
  */
 public final class DefaultGradleBuild implements GradleBuild {
 
-    private final IProject project;
+    private final Optional<org.eclipse.buildship.core.internal.workspace.GradleBuild> gradleBuild;
```

Comment:
❌ Never use Optional as a field type. We have `null` for that. This just creates unnecessary noise and garbage.

Reply:
Fixed.